test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/IVsServicesTestCase.cs

```diff
@@ -103,8 +97,11 @@ public async Task SimpleInstallFromIVsInstaller_PackageSourceMapping_WithSingleF
             await CommonUtility.CreatePackageInSourceAsync(privateRepositoryPath, packageName, packageVersion);
 
             // Create nuget.config with Package source mapping filtering rules before project is created.
-            CommonUtility.CreateConfigurationFile(Path.Combine(mainDirectory, "NuGet.config"), $@"<?xml version=""1.0"" encoding=""utf-8""?>
+            CommonUtility.CreateConfigurationFile(simpleTestPathContext.NuGetConfig, $@"<?xml version=""1.0"" encoding=""utf-8""?>
```

Comment:
You probably don't need this if you are using simple test path context.

Reply:
The code is setting up package source mappings in the config by just writing a static XML string to disk.  Do we have code in the SimpleTestPathContext/Settings stuff to add package source mappings?

Reply:
In this case it is needed, because it sets up package source mapping. SimpleTestPathContext sets up a nuget.config with a test specific global packages folder and package source, but I'm not aware of a way to set up package source mapping via APIs.

Reply:
https://github.com/NuGet/NuGet.Client/blob/20be995d651808feb5cd48f30cb97197e497377c/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestSettingsContext.cs#L100 can technically edit it. 
We can always add APIs in there that updates it. 
I don't think we should be using write to file.

Reply:
Okay I added a new API to add package source mappings in the test utility